proposals/awslogs.md

```diff
@@ -0,0 +1,195 @@
+    - The log stream will be named like this (by the ECS Agent): `prefix-name/container-name/ecs-task-id`
+
+*Solution:* Existing ability to configure logs remains undisturbed, but add additional flag `--create-logs` that creates the necessary log group(s) in CloudWatch.
+- The log configuration from the docker compose file will be read
+- If user has not specified a log stream prefix, warn them that we are auto-setting it to a default value in their task definition.
+    - *Additionally*, even if `--create-logs` is not specified, but we detect that the there is no prefix configured in their docker compose file (but log group and awslogs driver is specified), prefix will still be auto-set, and the user will be warned about this. This technically will break backwards compatibility- however, this risk is acceptable. It is very unlikely that ECS CLI users would actually desire to have their log streams named without a prefix. If no prefix is given, the ECS Agent sets the log stream name to be the container ID which was randomly generated by docker. Understanding this random ID requires logging into the underlying instances and retrieving info from the Docker Daemon. For all intents and purposes, the container ID is meaningless from a customer standpoint.
+    - The ECS CLI is designed to simplify workflows and make it easier to understand ECS. Therefore, we should be opinionated and protect users from accidentally configuring there logs in a poor way. We can help protect users from the less useful, complicated, legacy behavior of ECS.
+    - Additionally, the user will be warned that the default retention policy is to keep all log events forever, causing them to be charged for all time. They can change the policy in the CloudWatch Console or AWS CLI.
+
+```
+ecs-cli compose up --help
+--create-logs                     Creates any necessary log groups in CloudWatch.
+```
+
+```
+ecs-cli compose up --create-logs
+INFO[0000] Creating Resources in CloudWatch for your logs.
+WARN[0001] You have not specified a log stream prefix, auto-setting it to 'ecs-compose-'
+WARN[0002] By default, CloudWatch will store your logs forever, it is recommended that you set a retention policy.
+```
+
+*Suggested Configuration:*
+- If the user has not specified a log configuration in their compose file, then using the `--create-logs` command will fail and will print a help message with the suggested configuration. Here is one possible idea:
+For Services:
```

Comment:
I don't think we can currently override log configuration as part of the service definition, so if you want to do something different for services you'll need to keep track of which task definitions correspond to services and which ones are just used for one-off tasks.

Reply:
That's a good point, but to be clear, the idea is that the user has a compose file which they are using for their service, and only that service. We generally assume that ECS CLI users have all the resources for a task/service in a project directory and that this is not re-used for other tasks/services. So in the task definition that is used by their service, than can include the service name in the compose file.